### PR TITLE
python312Packages.dmgbuild: init at 1.6.2

### PR DESCRIPTION
**Pull request by Moraxyc**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Moraxyc** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.dmgbuild</li>
    <li>python311Packages.dmgbuild.dist</li>
    <li>python312Packages.dmgbuild</li>
    <li>python312Packages.dmgbuild.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/dmgbuild/default.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    repo = "dmgbuild";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-PozYxmXumFnptIgb4FM4b/Q5tx0MIS2bVw2kCuGucA8=";
+  };
+
+  postPatch = ''
+    # relax all deps
+    substituteInPlace pyproject.toml \
+      --replace-fail "==" ">="
+  '';
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/dmgbuild/default.nix:27]:
```suggestion
  pythonRelaxDeps = true;
```
You don't need the hook as it's python builder.

See https://nixos.org/manual/nixpkgs/unstable/#using-pythonrelaxdepshook

Comment (Moraxyc) [reply pkgs/development/python-modules/dmgbuild/default.nix:27]: `pythonRelaxDeps` is not work in this package.

https://github.com/dmgbuild/dmgbuild/blob/198eb9957ebb5b12e8524bb2d4dd63bc047c6594/pyproject.toml#L3

File: pkgs/development/python-modules/dmgbuild/default.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  # https://github.com/dmgbuild/dmgbuild/blob/refs/tags/v1.6.2/tests/examples/settings.py#L17
+  doCheck = false;
+
+  meta = {
+    description = "MacOS command line utility to build disk images";
+    homepage = "https://github.com/dmgbuild/dmgbuild";
+    changelog = "https://github.com/dmgbuild/dmgbuild/releases/tag/v${version}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ moraxyc ];
+    mainProgram = "dmgbuild";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/dmgbuild/default.nix:53]:
```suggestion
    mainProgram = "dmgbuild";
    platforms = lib.platforms.darwin;
```

Doesn't really build on other platforns; see ofborg build log

File: pkgs/development/python-modules/dmgbuild/default.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    platform = lib.platforms.darwin;
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/dmgbuild/default.nix:54]:
```suggestion
    platforms = lib.platforms.darwin;
```

Comment (Moraxyc) [reply pkgs/development/python-modules/dmgbuild/default.nix:54]: Sorry for that

**Review by Aleksanaa — APPROVED** [review]
Built on aarch64-darwin

**Pull request merged**